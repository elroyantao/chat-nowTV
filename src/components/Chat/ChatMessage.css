.ChatMessage {
  display: flex;
}

.ChatMessage-avatar {
  width: 100px;
  height: 100px;
}

ChatMessage-message {
  flex: 1;
}