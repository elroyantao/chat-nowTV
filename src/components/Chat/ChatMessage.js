import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import './ChatMessage.css';
import Avatar from '../User/Avatar'

class ChatMessage extends Component {
  static propTypes = {
    message: PropTypes.object.isRequired,
    user: PropTypes.object
  }
  render() {
    const { message, user = {} } = this.props
    return (
      <div title={user.email} className="ChatMessage">
        <div className="ChatMessage-avatar">
          <Avatar 
            avatar={user.avatar} 
            alt={user.firstName} 
          />
        </div>
        <div className="ChatMessage-message">{message.message}</div>
      </div>
    )
  }
}

const mapStateToProps = (state, props) => {
  return ({
    user: state.users[props.message.userId]
  })
}

export default connect(
  mapStateToProps,
  null
)(ChatMessage)