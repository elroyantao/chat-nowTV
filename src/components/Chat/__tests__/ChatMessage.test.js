import testHelper from '../../../helpers/test-helper'
import ChatMessage from '../ChatMessage'

describe('<ChatMessage/>', () => {
  const renderComponent = testHelper(ChatMessage.WrappedComponent)
  const initProps = {
    message: {
      id: 2348,
      message: 'Hi, this is a mocked message'
    }
  }
  describe('@renders', () => {
    it('in default state', () => {
      const { wrapper, getTree } = renderComponent(initProps)
      expect(getTree()).toMatchSnapshot()
      expect(wrapper.find('.ChatMessage').prop('title')).toBe(undefined)
      
    })
    it('when user is defined', () => {
      const { wrapper, getTree } = renderComponent({
        ...initProps,
        user: { 
          id: '0d9496c4-2490-40e6-a0c3-0964e371f1ab',
          firstName: 'Henry',
          lastName: 'Harrison',
          email: 'hharrisonj@github.io',
          avatar: 'http://dummyimage.com/100x100.bmp/cc0000/ffffff',
          ip: '125.12.50.203'
        }
      })
      expect(getTree()).toMatchSnapshot()
      expect(wrapper.find('.ChatMessage').prop('title')).toBe('hharrisonj@github.io')
    })
  })
})