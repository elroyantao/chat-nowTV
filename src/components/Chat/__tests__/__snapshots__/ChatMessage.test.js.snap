exports[`<ChatMessage/> @renders in default state 1`] = `
<div
  className="ChatMessage">
  <div
    className="ChatMessage-avatar">
    <Avatar />
  </div>
  <div
    className="ChatMessage-message">
    Hi, this is a mocked message
  </div>
</div>
`;

exports[`<ChatMessage/> @renders when user is defined 1`] = `
<div
  className="ChatMessage"
  title="hharrisonj@github.io">
  <div
    className="ChatMessage-avatar">
    <Avatar
      alt="Henry"
      avatar="http://dummyimage.com/100x100.bmp/cc0000/ffffff" />
  </div>
  <div
    className="ChatMessage-message">
    Hi, this is a mocked message
  </div>
</div>
`;
